startup/buildInfo.js

```diff
@@ -16,81 +16,167 @@ if (revId == null) throw new Error("No ADAPTABLE_APPREVISION_ID");
+        ];
+    } else if (tags.includes("python")) {
+        // Use the older builder for versions < 3.10
+        if (["3.6", "3.7", "3.8", "3.9"].includes(appConfig.pythonVersion || "")) {
+            imageBuildProps.config.builder = oldBuilderImage;
 
-touch __adaptable.py
-`;
+            // Workaround is only required for the older builder
+            imageBuildProps.config.preBuildScript = pythonWorkaround;
+        }
 
-if (tags.includes("nodejs")) {
-    // Use the older builder for versions < 18
-    if (["12", "14", "16"].includes(appConfig.nodeVersion || "")) {
-        imageBuildProps.config.builder = oldBuilderImage;
+        imageBuildProps.config.buildpacks = [
+            "paketo-buildpacks/python",
+            // buildpack-launch is required for BP_LAUNCH_COMMAND
+            "adaptable/buildpack-launch:0.0.7",
+        ];
+    }
+} else {
+    // Nixpacks
+
+    let providers;
+
+    if (tags.includes("nodejs")) providers = ["node"];
+    else if (tags.includes("python")) providers = ["python"];
+    else if (tags.includes("go")) providers = ["go"];
+
+    const setIfDefined = {
+        NIXPACKS_NODE_VERSION: appConfig.nodeVersion,
+        NIXPACKS_PYTHON_VERSION: appConfig.pythonVersion,
+        NIXPACKS_START_CMD: appConfig.startCommand,
+    };
+
+    const variables = {};
+
+    Object.entries(setIfDefined).forEach(([key, val]) => {
+        if (val != null) variables[key] = val;
+    });
+
+    /**
+     * @type Record<string, any>
+     */
+    const plan = {
+        providers,
+        variables,
+        phases: {},
+    };
 
-        // Older builder errors if BP_NODE_RUN_SCRIPTS is the empty string
-        if (!env.BP_NODE_RUN_SCRIPTS) delete env.BP_NODE_RUN_SCRIPTS;
+    // Install phase
+    if (appConfig.installCommand) {
+        plan.phases.install = {
+            cmds: [appConfig.installCommand],
+        };
     }
 
-    imageBuildProps.config.buildpacks = [
-        "paketo-buildpacks/nodejs",
-        // buildpack-launch is required for BP_LAUNCH_COMMAND
-        "adaptable/buildpack-launch:0.0.7",
-    ];
-} else if (tags.includes("python")) {
-    // Use the older builder for versions < 3.10
-    if (["3.6", "3.7", "3.8", "3.9"].includes(appConfig.pythonVersion || "")) {
-        imageBuildProps.config.builder = oldBuilderImage;
-
-        // Workaround is only required for the older builder
-        imageBuildProps.config.preBuildScript = pythonWorkaround;
+    // Build phase
+    if (appConfig.nodeRunScripts && appConfig.buildCommand) {
+        throw new Error(`Cannot specify both nodeRunScripts and buildCommand together`);
     }
 
-    imageBuildProps.config.buildpacks = [
-        "paketo-buildpacks/python",
-        // buildpack-launch is required for BP_LAUNCH_COMMAND
-        "adaptable/buildpack-launch:0.0.7",
-    ];
+    if (appConfig.nodeRunScripts) {
+        const cmds = appConfig.nodeRunScripts.split(",").map((s) => `npm run ${s}`);
```

Comment:
Is it always safe to use npm here, or do we have to use yarn if yarn is the default package manager?

Reply:
Nixpacks does install npm along with node, so this is always safe.